system/lib/pthread/pthread_join.c

```diff
@@ -9,8 +9,31 @@
 #include <pthread.h>
 
 extern int __pthread_join_js(pthread_t thread, void **retval);
+
 int __pthread_join(pthread_t thread, void **retval) {
+  if (thread->self != thread) {
```

Comment:
The JS version checked if `thread` was NULL before doing these, do we not still want that?

Reply:
None of the musl entry points to this.. so it seems unnecessary.. calling these functions with NULL in undefined behaviour,  I suppose we could at a check in debug builds, but we might also need to then add a debug configuration of libc.   I'm tempted to go with musl on this one.

Reply:
Fair enough. ASan should catch bugs there anyhow.